target/linux/generic/files/drivers/net/phy/ar8216.c

```diff
@@ -296,6 +297,28 @@ ar8xxx_rmw(struct ar8xxx_priv *priv, int reg, u32 mask, u32 val)
 	return ret;
 }
 
+u32
+ar8xxx_rmr(struct ar8xxx_priv *priv, int reg, u32 mask)
+{
+	struct mii_bus *bus = priv->mii_bus;
+	u16 r1, r2, page;
+	u32 ret;
+
+	split_addr((u32) reg, &r1, &r2, &page);
+
+	mutex_lock(&bus->mdio_lock);
+
+	bus->write(bus, 0x18, 0, page);
+	wait_for_page_switch();
+
+	ret = ar8xxx_mii_read32(priv, 0x10 | r2, r1);
+	ret &= mask;
+
+	mutex_unlock(&bus->mdio_lock);
+
+	return ret;
+}
```

Comment:
This function is rather useless, since it does the same thing as ar8xxx_read with the addition of returning ret & mask. Please remove it.